scripts/jenkins/cloud/ansible/roles/list_packages/tasks/diff-packages.yml

```diff
@@ -0,0 +1,55 @@
+#
+# (c) Copyright 2019 SUSE LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+#
+---
+
+- name: Diff list of installed packages in plaintext
+  shell: |
+    diff -U 0 /tmp/rpms/rpms_{{ item }}_{{ state1 }}.txt \
+      /tmp/rpms/rpms_{{ item }}_{{ state2 }}.txt \
+      > /tmp/rpms/diffs/rpms_{{ item }}.diff
+  args:
+    warn: False
+  register: diff_list
+  # diff rc: 0 when nothing changes(handled bellow) or 2 when error
+  failed_when: diff_list.rc == 2
+  with_items: "{{ [cloud_env]+groups['cloud_virt_hosts'] }}"
+
+
+- name: Concatenate plaintext diffs
+  run_once: true
+  shell: |
+    rm /tmp/rpms/diffs/rpms_cloud.diff
+    for f in `ls -1 /tmp/rpms/diffs/ | grep -v rpms_cloud.diff`; do
+        cat /tmp/rpms/diffs/$f >> /tmp/rpms/diffs/rpms_cloud.diff
+    done
+    cat /tmp/rpms/diffs/rpms_cloud.diff
+  register: rpms_cloud_diff
+  args:
+    warn: False
+
+- name: Check size of rpms_cloud.diff file
+  stat:
+    path: /tmp/rpms/diffs/rpms_cloud.diff
+  register: cloud_diff_file
+
+- name: Fail if no changed package versions found
```

Comment:
This should be a generic playbook/role, so you shouldn't fail if no packages have been updated. Also, expect to have no packages updated on some nodes, even if you have on other nodes (e.g. a `crowbar-openstack` package update will only be observed on the admin node).

Reply:
This will fail only when there is nothing changed on all nodes(rpms_cloud.diff is containing all the diffs available)

Reply:
Slightly re-done.